components/Tags.tsx

```diff
@@ -1,53 +1,97 @@
+import { isDefined } from "helpers/tinaHelpers";
 import { useEffect } from "react";
 import { resetPaginationPage } from "redux/slices/pagination";
 import { resetTags, setTags } from "redux/slices/selectedTags";
 import { useAppDispatch, useAppSelector } from "redux/typesHooks";
+import { useEditState } from "tinacms/dist/react";
+
+export interface Tag {
+  name: string;
+  postsCount: number;
+}
 
 interface Props {
-  uniqueTags: string[];
+  tags: Tag[];
+}
+
+interface TagItemProps {
+  isActive: boolean;
+  onClick: () => void;
+  tagName: string;
+  tagCount?: number;
 }
 
-const Tags = ({ uniqueTags }: Props) => {
+interface TagsListProps {
+  tags: TagItemProps[];
+}
+
+const MINIMUM_VISIBLE_COUNT = 2;
+const SINGLE_POST_COUNT = 1;
+
+const TagItem = ({
+  isActive,
+  onClick,
+  tagName,
+  tagCount,
+}: TagItemProps): JSX.Element => (
+  <li className={isActive ? "active" : ""} onClick={onClick}>
+    {tagCount ? `${tagName} [${tagCount}]` : tagName}
+  </li>
+);
+
+const TagsList = ({ tags }: TagsListProps): JSX.Element => (
+  <ul className="filter-tags-list">
+    {tags.map((tag) => (
+      <TagItem key={tag.tagName} {...tag} />
```

Comment:
```suggestion
      <TagItem key={tag.tagName} tag={tag} />
```

Feels like this is generally a safer approach